src/render/graph.rs

```diff
@@ -126,6 +125,7 @@ impl Graph {
         let inputs = vec![AudioRenderQuantum::from(self.alloc.silence()); number_of_inputs];
         let outputs = vec![AudioRenderQuantum::from(self.alloc.silence()); number_of_outputs];
 
+        let index = index.0 as usize;
```

Comment:
Hey, really interesting perf improvements
Just thinking (maybe I'm mistaking, not really sure and maybe that's what you have in mind but...) renaming these to `id` rather than `index` could help continue refactoring I think. 
Because then we could have two parallel `Vec`: a `Vec<Option<AudioNodeId>>` and a `Vec<Option<RefCell<Node>>>` where the indexes would correspond. Then we could reuse `None` places in the `Vec` to prevent the it from growing on indefinitely (would maybe imply a third `Vec<Option<usize>>` containing the dropped positions for housekeeping)

Reply:
Hey thanks for the suggestion. I actually have another solution in mind where the audio graph lets the AudioContext know ( realtime safe) which AudioNodeIds are available again. Would be interesting to benchmark against your solution